internal/exporter/prometheus/collectors/cpu_info.go

```diff
@@ -0,0 +1,193 @@
+// SPDX-FileCopyrightText: 2025 The Kepler Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package collectors
+
+import (
+	"bufio"
+	"os/exec"
+	"strings"
+	"sync"
+
+	prom "github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	architecture     = "architecture"
+	model_name       = "model_name"
+	cpus             = "cpus"
+	cores_per_socket = "cores_per_socket"
+	sockets          = "sockets"
+	vendor_id        = "vendor_id"
+)
+
+// cmdRunner defines an interface for running commands.
+type cmdRunner interface {
+	Run(name string, args ...string) ([]byte, error)
+}
+
+// execRunner implements CommandRunner using exec.Command.
+type execRunner struct{}
+
+func (e *execRunner) Run(name string, args ...string) ([]byte, error) {
+	cmd := exec.Command(name, args...)
+	return cmd.CombinedOutput()
+}
+
+type cpuInfo struct {
+	sync.Mutex
+	sync.Once
+	sysroot  string
+	desc     *prom.Desc
+	cache    map[string]string
+	cacheErr error
+
+	commandRunner cmdRunner
+}
+
+var _ prom.Collector = &cpuInfo{}
+
+// NewCpuInfoCollector creates a new CPU info collector.
+// The sysroot parameter specifies the path to the root filesystem (e.g., /host in a container),
+// allowing lscpu to read CPU information from the host's /proc and /sys when mounted.
+func NewCpuInfoCollector(sysroot string) *cpuInfo {
+	return &cpuInfo{
+		sysroot: sysroot,
+		desc: prom.NewDesc(
+			prom.BuildFQName(namespace, "", "cpu_info"),
+			"CPU information from lscpu command",
+			[]string{
+				architecture,
+				model_name,
+				cpus,
+				cores_per_socket,
+				sockets,
+				vendor_id,
+			},
+			nil,
+		),
+		cache:         make(map[string]string),
+		commandRunner: &execRunner{},
+	}
+}
+
+func (ci *cpuInfo) Describe(ch chan<- *prom.Desc) {
+	ch <- ci.desc
+}
+
+func (ci *cpuInfo) Collect(ch chan<- prom.Metric) {
+	ci.Lock()
+	defer ci.Unlock()
+
+	// Populate cpuinfo cache once
+	ci.Do(func() {
+		ci.cache, ci.cacheErr = ci.getCPUInfo()
+	})
+
+	// Skip metric emission if initialization failed
+	if ci.cacheErr != nil {
+		return
+	}
+
+	// Use cached values
+	ch <- prom.MustNewConstMetric(
+		ci.desc,
+		prom.GaugeValue,
+		1.0,
+		ci.cache[architecture],
+		ci.cache[model_name],
+		ci.cache[cpus],
+		ci.cache[cores_per_socket],
+		ci.cache[sockets],
+		ci.cache[vendor_id],
+	)
+}
+
+func (ci *cpuInfo) getCPUInfo() (map[string]string, error) {
+	kv, err := ci.runLscpu()
```

Comment:
suggestion: wondering if https://github.com/prometheus/procfs/blob/master/cpuinfo.go is simpler?

Reply:
this implementation is also not inventing anything new, uses a utility which is quite quite standard and architecture independent.

Reply:
let me know if this is not acceptable. i will close the PR and create a new based on your suggestion.

Reply:
happy to merge this but I do think we should consider not having a dependency (also consider security implications) especially because a package we depend on already provides this without any additional dependency.